src/components/table/index.js

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable no-nested-ternary */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  DefaultTable,
+  TableContainer,
+  Tr,
+  Td,
+  Th,
+  ArrowIcon,
+  HeaderContainer,
+  HeaderText,
+} from './components';
+
+const Table = ({
+  dataSource,
+  colNames,
+  headerColor,
+  variant,
+  sortIcon,
+  stripped,
+  hoverable,
+  bordered,
+  ...rest
+}) => {
+  const [obtaineddataSource, setobtaineddataSource] = useState(dataSource);
+  const [colSortIndex, setSortIndex] = useState([]);
+
+  const sortDataAscending = (key, index) => {
+    const newobtaineddataSource = [...obtaineddataSource];
+    newobtaineddataSource.sort((a, b) => {
+      let keyA = a[key];
+      let keyB = b[key];
+      if (typeof a[key] === 'string') {
+        keyA = keyA.toLowerCase();
+        keyB = keyB.toLowerCase();
+      }
```

Comment:
We can do: 
```
const keyA = String(a[key]).toLowerCase();
const keyB = String(b[key]).toLowerCase();
```
instead of 

```
let keyA = a[key];
      let keyB = b[key];
      if (typeof a[key] === 'string') {
        keyA = keyA.toLowerCase();
        keyB = keyB.toLowerCase();
      }
```

That would remove the chances of having any other types to process than string.